### ISSUE TITLE
unicorn/no-for-loop False positive

### ISSUE THREAD

**Issue by sk-** [sindresorhus/eslint-plugin-unicorn#829]
The following code triggers a `unicorn/no-for-loop`, however there's no straightforward way to refactor the code using Vanilla JS.

```js
const arr1 = [];
const arr2 = [];
for (let index = 0; index < arr1.length; index++) {
  arr2[index].name = arr1[index].original_name;
}
```

Note that when you assign in the loop to the same array, then no error is raised.

**Comment by sindresorhus** [comment]
There is. You can use `Array#entries()`.

**Comment by sk-** [comment]
Thanks, for future readers, the code should look like:

```js
for (const [index, element] of Object.entries(arr1)) {
  arr2[index].name = element.original_name;
}
```

**Comment by fisker** [comment]
`Object.entries` is different from `Array#entries`

Should be

```js
for (const [index, element] of arr1.entries()) {
  arr2[index].name = element.original_name;
}
```
